### PR TITLE
Handle system state messages

### PR DESCRIPTION
**Pull request by slevertbiot**

### CODE REVIEW

**Comment by slevertbiot** [comment]
I've altered this project in two significant ways.  It's now a typescript project and it's broken out into multiple source files.

From a CI/CD perspective, I believe we now need to create a zip file of the compiled js files in the dist folder and deploy the zip.  I'll raise a story for devops to address this.

File: udmif/event-handler/package.json [hunk 1]
```diff
@@ -2,21 +2,37 @@
...
   },
   "scripts": {
-    "watch": "npm-watch start",
-    "start": "functions-framework --target=handleUDMIEvent --signature-type=event"
+    "dev": "nodemon --watch src --watch data -e js,ts,txt,response,json --exec node -r ts-node/register --inspect=0.0.0.0:6329  src/index.ts",
+    "start": "functions-framework --source=dist/ --target=handleUdmiEvent --signature-type=event",
+    "watch": "concurrently \"tsc -w\" \"nodemon --watch ./build/ --exec npm run start\"",
+    "test": "NODE_ENV=test jest --runInBand --silent",
+    "testInteractive": "NODE_ENV=test jest --runInBand --silent --watchAll",
+    "postinstall": "cd ../.. && husky install udmif/api/.husky"
```
Comment (mspratt-biot) [inline udmif/event-handler/package.json:33]: Shouldn't this line be changed to be:
```
"postinstall": "cd ../.. && husky install udmif/event-handler/.husky"
```
You'll also need to add a `pre-commit` file in an `event-handler/.husky` folder like we've done in `web/.husky` and `api/.husky`. Don't worry about the `_` folder in `.husky`, that's autogenerated. The contents of the `pre-commit` file should be:
```
#!/bin/sh
. "$(dirname "$0")/_/husky.sh"

cd udmif/event-handler
npx lint-staged
```

File: udmif/event-handler/.gitignore [hunk 2]
```diff
@@ -22,3 +22,5 @@ setupEnv.sh
 .mongodb
 
 dist/
+
+.husky/_
```
Comment (mspratt-biot) [inline udmif/event-handler/.gitignore]: This doesn't need to be added here, the `postinstall` script will generate it's own `.gitignore` in the `.husky/` folder.

Comment (mspratt-biot) [reply udmif/event-handler/.gitignore]: Oh I see you did that in the next commit. Ok cool.

**Review by mspratt-biot — APPROVED** [review]

**Pull request merged**